components/popover/popover-mixin.js

```diff
@@ -125,25 +153,42 @@ export const PopoverMixin = superclass => class extends superclass {
 		this._dismissibleId = null;
 	}
 
-	_close() {
-		const hide = () => {
-			this.opened = false;
-		};
+	_focusContent(container) {
+		if (this.noAutoFocus || !this._applyFocus) return;
+
+		const focusable = getFirstFocusableDescendant(container);
+		if (focusable) {
+			// Removing the rAF call can allow infinite focus looping to happen in content using a focus trap
+			requestAnimationFrame(() => focusable.focus());
```

Comment:
If consumers wanted to call `open` or `close` and use the Promise to know when that operation were complete AND focus had been applied, how would they do that? For example in a visual diff scenario. Just wait for the event to fire?

Reply:
Currently yeah, they would need to wait for the event (perhaps on frame afterwards). I could potentially set up `open`, `close`, and `toggleOpen` to resolve their returned promises after that - that was some sugar that I sort of had in mind, but there's more that needs to be done to further develop those methods for other stories, so I've kept them to a bare minimum for now. Happy to add it now or later.

Reply:
I think it's OK as-is, as long as there's a consistent way to do it. Often we'll do something like:

```javascript
setTimeout(() => popover.open());
await oneEvent(popover, 'd2l-popover-open');
```

So as long as things are all focused and in a competed state by then, that seems good.